tests/mock_crust/utils.rs

```diff
@@ -344,13 +352,23 @@ pub fn create_connected_nodes_with_cache_till_split(network: &Network) -> Vec<Te
                     Event::NodeAdded(..) |
                     Event::Connected |
                     Event::Tick => (),
-                    Event::GroupSplit(..) => break 'outer,
+                    Event::GroupSplit(prefix) => {
+                        let _ = split_events.insert(prefix);
+                        if split_events.len() == split_count {
+                            break 'outer;
```

Comment:
Maybe it would be cleaner to just use
```rust
while split_events.len() < split_count
```
above. Also, the set of split prefixes doesn't seem to be needed, so you could just make `split_count` count _down_ to 0 instead?